packages/firebase_ml_vision/android/src/main/java/io/flutter/plugins/firebasemlvision/FirebaseMlVisionPlugin.java

```diff
@@ -39,10 +40,24 @@ public static void registerWith(Registrar registrar) {
 
   @Override
   public void onMethodCall(MethodCall call, Result result) {
-    if (call.method.equals("TextDetector#detectInImage")) {
-      handleTextDetectionResult(call, result);
-    } else {
-      result.notImplemented();
+    switch (call.method) {
+      case "TextDetector#detectInImage":
+        handleTextDetectionResult(call, result);
+        break;
+      case "TextDetector#close":
+        if (textDetector != null) {
+          try {
+            textDetector.close();
+          } catch (IOException exception) {
+            result.error("textDetectorError", exception.getLocalizedMessage(), null);
```

Comment:
If we get to this code, we'll be invoking both `result.error` and `result.success`.